translate api/application.md (#845)
vuejs-translations/docs-ja: src/api/application.md

  const app = createApp(App)
  ```

- **参照:** [ガイド - Vue アプリケーションの作成](/guide/essentials/application.html)

## createSSRApp()

> [SSR hydration (ハイドレーション) ](/guide/scaling-up/ssr.html#client-hydration)モードでアプリケーションインスタンスを作成します。使用法は `createApp()` と全く同じです。

Reviewer: リンク修正お願いします:pray:
```suggestion
[SSR hydration (ハイドレーション) ](/guide/scaling-up/ssr.html#クライアント・ハイドレーション)モードでアプリケーションインスタンスを作成します。使用法は `createApp()` と全く同じです。
```

## app.mount()

コンテナ要素にアプリケーションインスタンスをマウントします。

- **型**

  ```ts
  interface App {
    mount(rootContainer: Element | string): ComponentPublicInstance
  }
  ```

- **詳細**

  引数は、実際の DOM 要素または CSS セレクターのいずれかにすることができます (最初にマッチした要素を使用します)。ルートコンポーネントのインスタンスを返します。

  コンポーネントにテンプレートがあるか、render 関数を定義している場合、コンテナ内の既存の DOM ノードを置換します。それ以外の場合は、実行時コンパイラーが使用可能であれば、コンテナの `innerHTML` がテンプレートとして使用されます。

>   SSR ハイドレーションモードでは、コンテナ内の既存の DOM ノードをハイドレートします。[ミスマッチ](/guide/scaling-up/ssr.html#hydration-mismatch)があった場合、既存の DOM ノードは期待される結果に合うよう変化されます。

Reviewer: ```suggestion
  SSR ハイドレーションモードでは、コンテナ内の既存の DOM ノードをハイドレートします。[ミスマッチ](/guide/scaling-up/ssr.html#ハイドレーション・ミスマッチ)があった場合、既存の DOM ノードは期待される結果に合うよう変化されます。
```

  app.mount('#app')
  ```

  実際の DOM 要素にマウントすることもできます:

  ```js
  app.mount(document.body.firstChild)
  ```

## app.unmount()

> マウントしたアプリケーションインスタンスをアンマウトし、アプリケーションのコンポーネントツリーにあるすべてのコンポーネントに対して、ライフサイクルフックの unmount を呼び出します。

Reviewer: ```suggestion
マウントしたアプリケーションインスタンスをアンマウントし、アプリケーションのコンポーネントツリーにあるすべてのコンポーネントに対して、ライフサイクルフックの unmount を呼び出します。
```

  </div>

- **参照:**
  - [Provide / Inject](/guide/components/provide-inject.html)

>   - [アプリケーションレベルの Provide ](/guide/components/provide-inject.html#app-level-provide)

Reviewer: ```suggestion
  - [アプリケーションレベルの Provide](/guide/components/provide-inject.html#アプリケーションレベルの-provide)
```

  app.use(MyPlugin)
  ```

- **参照:** [プラグイン](/guide/reusability/plugins.html)

## app.mixin()

グローバルミックスイン (スコープがアプリケーション) を適用します。グローバルミックスインは、インクルードされたオプションをアプリケーションの全コンポーネントインスタンスに適用します。

:::warning 非推奨

> ミックインはエコシステムのライブラリーで幅広く使用していることから、主に後方互換性ため、Vue 3 でサポートしています。ミックイン (特にグローバルミックスイン) はアプリケーションコードでは避けるべきです。

Reviewer: ```suggestion
ミックインはエコシステムのライブラリーで幅広く使用されていることから、主に後方互換性のため、Vue 3 でサポートしています。ミックイン (特にグローバルミックスイン) はアプリケーションコードでは避けるべきです。
```

グローバルミックスイン (スコープがアプリケーション) を適用します。グローバルミックスインは、インクルードされたオプションをアプリケーションの全コンポーネントインスタンスに適用します。

:::warning 非推奨

> ミックインはエコシステムのライブラリーで幅広く使用されていることから、主に後方互換性ため、Vue 3 でサポートしています。ミックイン (特にグローバルミックスイン) はアプリケーションコードでは避けるべきです。

Reviewer: 前回 suggestion の修正漏れ
```suggestion
ミックインはエコシステムのライブラリーで幅広く使用されていることから、主に後方互換性のため、Vue 3 でサポートしています。ミックイン (特にグローバルミックスイン) はアプリケーションコードでは避けるべきです。
```

## app.version

アプリケーションを作成した Vue のバージョンを提供します。Vue のバージョンの違いに応じた条件付きロジックが必要となる[プラグイン](/guide/reusability/plugins.html)内で有用となります。

- **型**

  ```ts
  interface App {
    version: string
  }
  ```

- **例**

>   プラグイン内でバージョンチェックを行います。

Reviewer: ```suggestion
  プラグイン内でバージョンチェックを行います:
```

## app.config.errorHandler

アプリケーション内から伝播する未捕捉のエラーに対して、グローバルハンドラーを割り当てます。

- **型**

  ```ts
  interface AppConfig {
    errorHandler?: (
      err: unknown,
      instance: ComponentPublicInstance | null,
      // `info` は Vue 特有のエラー情報です。

>       // 例 エラーがthrowされたのはどのライフサイクルフックか

Reviewer: ```suggestion
      // 例 エラーが throw されたのはどのライフサイクルフックか
```

      info: string
    ) => void
  }
  ```

- **詳細**

  エラーハンドラーは、エラー、エラーを起こしたコンポーネントインスタンス、エラーのソースの種類を指定した情報の文字列という 3 つの引数を受け取ります。

>   以下のソースから、エラーを捕捉することができます。

Reviewer: ```suggestion
  以下のソースから、エラーを捕捉することができます:
```

  }
  ```

- **詳細**

>   警告ハンドラーは、第 1 引数として警告メッセージ、第 2 引数としてソースコンポーネントのインスタンス、第 3 引数としてコンポーネントトーレスの文字列を受け取ります。

Reviewer: ```suggestion
  警告ハンドラーは、第 1 引数として警告メッセージ、第 2 引数としてソースコンポーネントのインスタンス、第 3 引数としてコンポーネントトレースの文字列を受け取ります。
```

  特定の警告をフィルタリングし、コンソールの詳細度を下げるために、使用することもできます。すべての Vue の警告は開発中に対応すべきで、多くの警告の中で特定の警告に集中するのはデバッグ期間のみ推奨されており、デバッグ完了時には解除すべきです。

  :::tip
  警告は開発環境でのみ動作しますので、この設定はプロダクションモードでは無視されます。
  :::

- **例**

  ```js
  app.config.warnHandler = (msg, instance, trace) => {
    // `trace` はコンポーネント階層のトレースです。
  }
  ```

## app.config.performance

> これを `true` に設定することで、ブラウザーの開発ツールの performance/timeline パネルで、コンポーネントの初期化、コンパイル、レンダリング、パッチについてのパフォーマンスのトレースが有効となります。 開発モードおよび [performance.mark](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) API をサポートするブラウザーでのみ動作します。

Reviewer: 前回見落としていましたが、日本語版のリンクに変更お願いします:pray:
```suggestion
これを `true` に設定することで、ブラウザーの開発ツールの performance/timeline パネルで、コンポーネントの初期化、コンパイル、レンダリング、パッチについてのパフォーマンスのトレースが有効となります。 開発モードおよび [performance.mark](https://developer.mozilla.org/ja/docs/Web/API/Performance/mark) API をサポートするブラウザーでのみ動作します。
```

> これを `true` に設定することで、ブラウザーの開発ツールの performance/timeline パネルで、コンポーネントの初期化、コンパイル、レンダリング、パッチについてのパフォーマンスのトレースが有効となります。 開発モードおよび[ performance.mark ](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) API をサポートするブラウザーでのみ動作します。

Reviewer: ```suggestion
これを `true` に設定することで、ブラウザーの開発ツールの performance/timeline パネルで、コンポーネントの初期化、コンパイル、レンダリング、パッチについてのパフォーマンスのトレースが有効となります。 開発モードおよび [performance.mark](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) API をサポートするブラウザーでのみ動作します。
```

- **型**: `boolean`

- **参照:** [ガイド - パフォーマンス](/guide/best-practices/performance.html)

## app.config.compilerOptions

実行時コンパイラーのオプションを設定します。このオブジェクトに設定される値はブラウザー内のテンプレートコンパイラーに渡され、設定されたアプリケーションのコンポーネントすべてに影響を与えます。[`compilerOptions` オプション](/api/options-rendering.html#compileroptions)を使用することで、コンポーネント毎のオプションを上書きすることも可能となることに注意してください。

::: warning 重要
フルビルド (則ち、ブラウザのテンプレートをコンパイルできるスタンドアロンの `vue.js`) 使用時のみ、この設定オプションが反映されます。ビルド設定で、実行時のみビルドを使用している場合は代わりに、ビルドツール設定により、`@vue/compiler-dom` にコンパイラオプションを渡さなければなりません。

> - `vue-loader` の場合: [`compilerOptions` ローダオプションによって渡す](https://vue-loader.vuejs.org/options.html#compileroptions). 参照 [`vue-cli` での設定方法](https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader).

Reviewer: ```suggestion
- `vue-loader` の場合: [`compilerOptions` ローダーオプションによって渡す](https://vue-loader.vuejs.org/options.html#compileroptions)。[`vue-cli` での設定方法](https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader)も参照してください。
```

- `vite` の場合: [`@vitejs/plugin-vue` オプションによって渡す](https://github.com/vitejs/vite/tree/main/packages/plugin-vue#options).
  :::

### app.config.compilerOptions.isCustomElement

> ネイティブカスタム要素を認識するため、チェックメソッドを指定する。

Reviewer: ```suggestion
ネイティブカスタム要素を認識するためのチェックメソッドを指定する。
```

- **例**

  ```js
  app.config.compilerOptions.whitespace = 'preserve'
  ```

### app.config.compilerOptions.delimiters

テンプレート内のテキスト補間に用いるデリミタを調整します。

- **型:** `[string, string]`

- **デフォルト:** `{{ "['\u007b\u007b', '\u007d\u007d']" }}`

- **詳細**

  これは通常、マスタッシュ記法も使用するサーバサイドのフレームワークとの衝突を回避するために使用します。

- **例**

  ```js
  // デリミタを、ES6 テンプレートの文字列スタイルに変更する。
  app.config.compilerOptions.delimiters = ['${', '}']
  ```

### app.config.compilerOptions.comments

テンプレートの HTML コメントの扱い方を調整します。

- **型:** `boolean`

- **デフォルト:** `false`

- **詳細**

  デフォルトでは、Vue はプロダクションでコメントを除去します。このオプションを `true` にすることで、プロダクションでもコメントを保持するよう Vue に強制することができます。開発ではコメントは常に保持されます。このオプションは通常、HTML コメントに依存するその他のライブラリーで Vue を使用する場合に使用します。

- **例**

  ```js
  app.config.compilerOptions.comments = true
  ```

## app.config.globalProperties

アプリケーション内部のコンポーネントインスタンスにアクセス可能なグローバルプロパティを登録するのに使用することができるオブジェクトです。

- **型**

  ```ts
  interface AppConfig {
    globalProperties: Record<string, any>
  }
  ```

- **詳細**

  これは、Vue 3 でなくなった Vue 2 の `Vue.prototype` の代替です。グローバルなものについてはすべて、慎重に使用すべきです。

  グローバルプロパティがコンポーネント自身のプロパティと衝突する場合、コンポーネント自身のプロパティの方が優先度が高くなります。

- **使用法**

  ```js
  app.config.globalProperties.msg = 'hello'
  ```

>   これにより、アプリケーションのコンポーネントテンプレート内およびコンポーネントインスタンスの `this` でも、`msg` を使用できるようになります。

Reviewer: ```suggestion
  これにより、アプリケーションのコンポーネントテンプレート内およびコンポーネントインスタンスの `this` でも、`msg` を使用できるようになります:
```

  type OptionMergeFunction = (to: unknown, from: unknown) => any
  ```

- **詳細**

  プラグインやライブラリーには、(グローバルミックインを注入することにより) カスタムコンポーネントのオプションのサポートを追加しているものもあります。複数のソースから同じオプションを "マージ" する必要がある場合 (たとえば、ミックスインやコンポーネントの継承)、これらのオプションには特別なマージロジックが必要となる場合があります。

  オプション名をキーとして使用し、`app.config.optionMergeStrategies` オブジェクトを割り当てることで、カスタムオプションにマージ戦略関数を登録することができます。

  マージ戦略関数は、第一引数と第二引数としてそれぞれ、親と子に定義されたオプションの値を受け取ります。

- **例**

  ```js
  const app = createApp({

>     // self からのオプション

Reviewer: ```suggestion
    // 自身のオプション
```

    msg: 'Vue',
    // ミックスインからのオプション
    mixins: [
      {
        msg: 'Hello '
      }
    ],
    mounted() {

>       // this.$options に公開されているマージオプション

Reviewer: ```suggestion
      // this.$options に公開されているマージされたオプション
```
